sympy/sets/handlers/issubset.py

```diff
@@ -24,35 +26,46 @@ def is_subset_sets(a, b): # noqa:F811
     if (b.right_open and not a.right_open and fuzzy_bool(Eq(a.end, b.end))):
         return False
 
+
 @dispatch(Interval, FiniteSet)  # type: ignore # noqa:F811
 def is_subset_sets(a_interval, b_fs): # noqa:F811
     # An Interval can only be a subset of a finite set if it is finite
     # which can only happen if it has zero measure.
     if fuzzy_not(a_interval.measure.is_zero):
         return False
 
+
 @dispatch(Interval, Union)  # type: ignore # noqa:F811
 def is_subset_sets(a_interval, b_u): # noqa:F811
     if all(isinstance(s, (Interval, FiniteSet)) for s in b_u.args):
         intervals = [s for s in b_u.args if isinstance(s, Interval)]
+        if (a_interval == S.Reals and all(
+                i.start.is_finite and i.start.is_Number or
+                i.end.is_finite and i.end.is_Number
+                for i in intervals)):
+            # there are no ambiguous boundaries in Union so
+            # guaranteed gaps
+            return False
```

Comment:
Fuzzy logic routines like this should not depend on the effects of evaluation. The routine also needs to work correctly for `Union(..., evaluate=False)`.

Reply:
```
In [5]: Interval(-oo, oo).is_subset(Union(Interval(-oo, 2),Interval(2,oo), evaluate=False))
Out[5]: False

In [6]: Interval(-oo, oo).is_subset(Union(Interval(-oo, 2),Interval(2,oo)))
Out[6]: True
```

Reply:
OK, so we need a linear (fast) test. What about the recent commit?

Reply:
It seems very complicated compared to the test cases. What about something like:

1. Check all intervals have Number start and end (otherwise do nothing)
2. Sort intervals by start
3. Loop through comparing the ends (and `left_open/right_open`)

Reply:
I am guessing that in spirit of speed that we should not do so on the basis of `2`. And if something like `is_zero` should not use re-evaluation to tell if something *would* be zero if it evaluated, we should not use evaluation to know if a set has any joining intervals that might span the interval that we are testing for inclusion. So other than do the obvious "interval is left or right of all others or never overlaps with any element" that we should not do more. So the first part that I added which looks for two semi-infinite spanning intervals should be undone.

But we could add a test of whether the inf of the interval is left of all others or the sup is right of all others which would indicate that it is larger on an end than any other in the set and thus not contained.